### PR TITLE
PlatformConfig: automatically get platform_str from the enum name

### LINKED ISSUE #31843: [100$ bounty] get platform_str from the enum name
we currently have two separate names for each platform in openpilot, for example

`BRONCO_SPORT_MK1` (enum name)
`"FORD BRONCO SPORT 1ST GEN"` (platform_str)

this adds unnecessary complexity and an extra line to every platform config.

this bounty is for removing the platform string, and instead having every PlatformConfig have the platform_str of it's enum name automatically. i.e.

```python
BRONCO_SPORT_MK1 = FordPlatformConfig(
  FordCarInfo("Ford Bronco Sport 2021-23"),
  CarSpecs(mass=1625, wheelbase=2.67, steerRatio=17.7),
)

assert (BRONCO_SPORT_MK1.platform_str == "BRONCO_SPORT_MK1")
```

we must also update the [migration table](https://github.com/commaai/openpilot/blob/master/selfdrive/car/fingerprints.py#L54) to reflect these updates

### PR DESCRIPTION
**Pull request by thenhnn**
PlatformConfig.platform_str is now inherited from PlatformConfig's enum name.
platform_str is removed from PlatformConfigs in car/<vendor> directories. (i'm not sure that it is not used in tests somewhere)
Migration table is filled to not break compatibility.

Closes #31843

### CODE REVIEW

File: selfdrive/car/__init__.py [hunk 1]
```diff
@@ -261,14 +261,15 @@ def override(self, **kwargs):
...
 class PlatformConfig(Freezable):
-  platform_str: str
   car_info: list[CarDocs]
   specs: CarSpecs
 
   dbc_dict: DbcDict
 
   flags: int = 0
 
+  platform_str: str | None = None
```
Comment (adeebshihadeh) [inline selfdrive/car/__init__.py:271]: why can this be None now?

File: selfdrive/car/chrysler/values.py [hunk 2]
```diff
@@ -63,33 +58,28 @@ class CAR(Platforms):
...
 
   JEEP_GRAND_CHEROKEE_2019 = ChryslerPlatformConfig(  # includes 2020 Trailhawk
-    "JEEP GRAND CHEROKEE 2019",
     [ChryslerCarDocs("Jeep Grand Cherokee 2019-21", video_link="https://www.youtube.com/watch?v=jBe4lWnRSu4")],
     JEEP_GRAND_CHEROKEE.specs,
   )
 
   # Ram
   RAM_1500 = ChryslerPlatformConfig(
-    "RAM 1500 5TH GEN",
```
Comment (adeebshihadeh) [inline selfdrive/car/chrysler/values.py:86]: let's add 5TH_GEN to the enum name

File: selfdrive/car/chrysler/values.py [hunk 3]
```diff
@@ -63,33 +58,28 @@ class CAR(Platforms):
...
     [ChryslerCarDocs("Ram 1500 2019-24", car_parts=CarParts.common([CarHarness.ram]))],
     ChryslerCarSpecs(mass=2493., wheelbase=3.88, steerRatio=16.3, minSteerSpeed=14.5),
     dbc_dict('chrysler_ram_dt_generated', None),
   )
   RAM_HD = ChryslerPlatformConfig(
```
Comment (adeebshihadeh) [inline selfdrive/car/chrysler/values.py:82]: same here

File: selfdrive/car/fingerprints.py [hunk 4]
```diff
@@ -117,4 +126,224 @@ def all_legacy_fingerprint_cars():
   "HYUNDAI TUCSON HYBRID 4TH GEN": HYUNDAI.TUCSON_4TH_GEN,
   "KIA SPORTAGE HYBRID 5TH GEN": HYUNDAI.KIA_SPORTAGE_5TH_GEN,
   "KIA SORENTO PLUG-IN HYBRID 4TH GEN": HYUNDAI.KIA_SORENTO_HEV_4TH_GEN,
+  # Removal of platform_str
```
Comment (adeebshihadeh) [inline selfdrive/car/fingerprints.py:129]:
```suggestion

  # Removal of platform_str, see https://github.com/commaai/openpilot/pull/31868/
```

File: selfdrive/car/fingerprints.py [hunk 5]
```diff
@@ -117,4 +126,224 @@ def all_legacy_fingerprint_cars():
...
+  "COMMA BODY": BODY.BODY,
+  # Chrysler
```
Comment (adeebshihadeh) [inline selfdrive/car/fingerprints.py:131]: these brand name comments should be removed. they don't add anything

**Review by adeebshihadeh — COMMENTED** [review]
This is a huge improvement - very excited to get it in!

File: selfdrive/car/__init__.py (hunk 1 above)
Comment (thenhnn) [reply selfdrive/car/__init__.py:271]: platform_str is None when PlatformConfig is not in enum.

For example:

```python
config = ChryslerPlatformConfig(
  [ChryslerCarDocs("Ram 1500 2019-24", car_parts=CarParts.common([CarHarness.ram]))],
  ChryslerCarSpecs(mass=2493., wheelbase=3.88, steerRatio=16.3, minSteerSpeed=14.5)
 )
 assert config.platform_str == None
 ```

Comment (jnewb1) [reply selfdrive/car/__init__.py:250]: can it just raise an exception instead?

Comment (thenhnn) [reply selfdrive/car/__init__.py:250]: > can it just raise an exception instead?

i can try to implement this by overriding `getattr` in metaclass, but i'm not sure is it worth it.

**Comment by jnewb1** [comment]
looks like the tests are failing. you can quickly run the car tests with

`pytest selfdrive/car/tests/test_car_interfaces.py`

**Comment by thenhnn** [comment]
@jnewb1 should I mark pr as draft until i fix all tests?

**Comment by jnewb1** [comment]
> @jnewb1 should I mark pr as draft until i fix all tests?

yea, non-drafts should be ready to review and merge. looks close though!

**Comment by thenhnn** [comment]
@jnewb1 

> FAILED selfdrive/test/process_replay/test_regen.py::TestRegen::test_engaged_0_PRIUS_C2 - AssertionError: False is not true : openpilot not engaged in PRIUS_C2

How to debug this properly?

**Comment by jnewb1** [comment]
```diff
def migrate_carParams(lr, old_logtime=False):
  all_msgs = []
  for msg in lr:
    if msg.which() == 'carParams':
      CP = messaging.new_message('carParams')
      CP.valid = True
      CP.carParams = msg.carParams.as_builder()
      for car_fw in CP.carParams.carFw:
        car_fw.brand = CP.carParams.carName
      if old_logtime:
        CP.logMonoTime = msg.logMonoTime
+++   CP.carParams.carFingerprint = MIGRATION.get(CP.carParams.carFingerprint, CP.carParams.carFingerprint)
      msg = CP.as_reader()
    all_msgs.append(msg)

  return all_msgs
```

this line should fix proces replay

**Comment by jnewb1** [comment]
> carParams.carFingerprint: 2

only replay diff is the carFingerprint

**Comment by jnewb1** [comment]
If this looks good @sshane I'll look at fixing xx

File: selfdrive/car/__init__.py [hunk 6]
```diff
@@ -259,10 +260,17 @@ def init(self):
 
   def __post_init__(self):
     self.init()
-    self.freeze()
+
```
Comment (jnewb1) [inline selfdrive/car/__init__.py:263]:
```suggestion


```

**Comment by adeebshihadeh** [comment]
Let’s just merge and revert if we don’t like it!

**Comment by jnewb1** [comment]
I like the clean solution you found for this! would you like to take a look at https://github.com/commaai/openpilot/issues/31713?

**Pull request merged**